### PR TITLE
MAINT: Remove duplicate docstring and correct location of `__all__` in np.random

### PR DESCRIPTION
**Pull request by eric-wieser**
This should fix a lot of LGTM errors

Note that this leaves the value of `np.random.__doc__` unchanged

### CODE REVIEW

**Comment by eric-wieser** [comment]
Seems LGTM doesn't compute the alert diff correctly

**Comment by eric-wieser** [comment]
@jhelie, any ideas why?

**Comment by jhelie** [comment]
Hi guys, sorry to read about this. Unfortunately I’m away from the office and with a very limited connection so can’t investigate but @markshannon or @sjvs should be able to help.

**Comment by eric-wieser** [comment]
@charris:

The info files were used by `PackageLoader`, but that didn't work anymore anyway (and we just removed it)

> doc/summarize.py uses the info file

The only mention of `.info` in that file is instructions to ignore it. Either way, all this removes is `info.__all__`, which LGTM correctly reported as being invalid.

In future, I think we should drop all the `info` files.

**Comment by eric-wieser** [comment]
For now, I think the approach should be to move all the information out of `info` modules (and then import it into `info` as I do here) to the point that we can emit a deprecation warning when they're imported without it being triggered by an import of `numpy`.

**Comment by sjvs** [comment]
Thanks for pinging me @jhelie.

The Python team are looking into this at the moment. @markshannon will keep you posted.

**Comment by eric-wieser** [comment]
@sjvs: Would it ease debugging for you if this PR is left open?

**Comment by sjvs** [comment]
Thanks @eric-wieser, but that's not necessary. Go ahead and merge :slightly_smiling_face:!

**Comment by markshannon** [comment]
The reason we are still marking the names in `numpy.random.__init__.__all__` as not being defined is that most of the functions are defined in the Cython module `np.random.mtrand` but that module's location changes when numpy is built. 

In the source code, `numpy/random/mtrand` is a folder, but once built, `numpy.random.mtrand` is not a package but a Cython module. On my machine:
```python
>>> os.listdir(os.path.dirname(numpy.random.__file__))
['info.py', 'setup.py', 'mtrand.cpython-35m-x86_64-linux-gnu.so', '__init__.py', 'randomkit.h', 'tests', '__pycache__']
```
We analyse the code as it is laid out in the checkout, so our analysis gets confused by `numpy.random.mtrand`.

**Comment by sjvs** [comment]
Here's the PR that tweaks the LGTM.com Python analysis using an `lgtm.yml` file: https://github.com/numpy/numpy/pull/11738

File: numpy/random/__init__.py [hunk 1]
```diff
@@ -6,17 +6,15 @@
...
 ==============================================================================
-random               Uniformly distributed values of a given shape.
+random_sample        Uniformly distributed floats over ``[0, 1)``.
+random               Alias for `random_sample`.
 bytes                Uniformly distributed random bytes.
 random_integers      Uniformly distributed integers in a given range.
-random_sample        Uniformly distributed floats in a given range.
-random               Alias for random_sample
-ranf                 Alias for random_sample
-sample               Alias for random_sample
```
Comment (charris) [inline numpy/random/__init__.py]: I assume you removed this because it is not exported in `__all__`.

**Comment by charris** [comment]
Thanks Eric.

File: numpy/random/__init__.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/random/__init__.py]: Nope, I removed this simply because it was already absent from `random.__doc__`

**Pull request merged**